### PR TITLE
blade-formatter: init at 1.41.1

### PR DESCRIPTION
**Pull request by lelgenio**
## Description of changes

A formatter for Blade templates from Laravel PHP framework.

https://github.com/shufo/blade-formatter

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review pr 260000` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>blade-formatter</li>
  </ul>
</details>

### CODE REVIEW

File: pkgs/by-name/bl/blade-formatter/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          base = writeText "base" ''
+            @if(   true )  Hello world!   @endif
+          '';
+        } ''
+        blade-formatter $base > $out
+      '';
+    };
+  };
+
+  meta = with lib; {
```
Comment (heisfer) [inline pkgs/by-name/bl/blade-formatter/package.nix:55]: Remove `with lib` it's antipattern. Use `with lib.maintainers` etc below.

File: pkgs/by-name/bl/blade-formatter/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+, blade-formatter
+}:
+
+mkYarnPackage rec {
+  pname = "blade-formatter";
+  version = "1.38.2";
+
+  src = fetchFromGitHub {
+    owner = "shufo";
+    repo = pname;
```
Comment (heisfer) [inline pkgs/by-name/bl/blade-formatter/package.nix:17]: Replace it with "blade-formatter".

**Comment by lelgenio** [comment]
Things done:

- applied suggestions
- changed from `mkYarnPackage` to `buildNpmPackage`
- added an update script
- updated from `1.38.2` to `1.41.1`

**Comment by heisfer** [comment]
Result of `nixpkgs-review pr 260000` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>blade-formatter</li>
  </ul>
</details>

**Comment by heisfer** [comment]
Just a few additional notes.
1. `meta.mainProgram = "blade-formatter";` (so that it can be used with `lib.getExe pkgs.blade-formatter` instead of `lib.getExe' pkgs blade-formatter "blade-formatter"`
2. format with [nixfmt-rfc-style](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=nixfmt-rfc) (seems to be the new trend)


Tested basic blade file and worked as expected.

```
<x-layout> <x-slot:heading>Job</x-slot:heading> <h2 class="font-bold text-lg">{{ $job->title }}</h2> <p>Salary: {{$job->salary}}</p> @can('edit', $job) <p class="mt-6"> <x-button href="/jobs/{{ $job->id }}/edit">Edit Job</x-button> </p> @endcan </x-layout>
```
```
<x-layout> <x-slot:heading>Job</x-slot:heading>
    <h2 class="font-bold text-lg">{{ $job->title }}</h2>
    <p>Salary: {{ $job->salary }}</p> @can('edit', $job)
        <p class="mt-6"> <x-button href="/jobs/{{ $job->id }}/edit">Edit Job</x-button> </p>
    @endcan
</x-layout>
```

**Review by heisfer — APPROVED** [review]

**Review by heisfer — APPROVED** [review]
LGTM

**Comment by lelgenio** [comment]
Oops, fumbled `meta.platforms` :confounded:

**Comment by lelgenio** [comment]
Result of `nixpkgs-review pr 260000` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>blade-formatter</li>
  </ul>
</details>

File: pkgs/by-name/bl/blade-formatter/update.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p ripgrep common-updater-scripts prefetch-npm-deps jq
```
Comment (pbsds) [inline pkgs/by-name/bl/blade-formatter/update.sh:2]:
```suggestion
#!nix-shell -i bash -p ripgrep common-updater-scripts prefetch-npm-deps jq sd
```

Comment (lelgenio) [reply pkgs/by-name/bl/blade-formatter/update.sh]: Thanks :smiley: :+1:

**Comment by pbsds** [comment]
Congrats on the round PR number btw :tada:  :tada:  :tada:

**Pull request merged**